unstructured_inference/config.py

```diff
@@ -7,13 +7,26 @@
 """
 
 import os
+import tempfile
 from dataclasses import dataclass
+from functools import lru_cache
+from pathlib import Path
+
+
+@lru_cache(maxsize=1)
+def _get_tempdir(dir: str) -> str:
```

Comment:
one thing - `dir` is reserved as std function name. Just replace with `dirname` or smth.